File: coresuite-dotnet-standard-vb/generator/SimpleXMLReport.vb
Imports System
Imports System.Xml
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.PageElements

Namespace coresuite_dotnet_standard_vb.generator

    Public Class SimpleXMLReport
        'Template for document elements
        Public Shared MyTemplate As ceTe.DynamicPDF.Template = New ceTe.DynamicPDF.Template()

        'Page Dimensions of MyPages
        Public Shared MyPageDimensions As PageDimensions = New PageDimensions(PageSize.Letter, PageOrientation.Portrait, 54.0F)
        'Current MyPage that elements are being added to
        Public Shared CurrentPage As Page = Nothing
        'Top Y coordinate for the body of the report
        Public Shared BodyTop As Single = 38
        'Bottom Y coordinate for the body of the report
        Public Shared BodyBottom As Single = MyPageDimensions.Body.Bottom - MyPageDimensions.Body.Top
        'Current Y coordinate where elements are being added
        Public Shared CurrentY As Single = 0
        'Used to control the alternating background
        Public Shared AlternateBG As Boolean = False

        Public Shared Sub Run(outputPdfPath As String)
            'Create a MyDocument and set it's properties
            Dim MyDocument As Document = New Document With {
                .Creator = "SimpleXMLReport.aspx",
                .Author = "ceTe Software",
                .Title = "Simple XML Report"
            }

            Dim Reader As XmlTextReader = New XmlTextReader(Util.GetResourcePath("Data/ProductList.xml"))

            'Adds elements to the header groups
            SetTemplate()

            ' Add the template to the MyDocument
            MyDocument.Template = MyTemplate

            'Builds the report
            BuildDocument(MyDocument, Reader)

            'Outputs the SimpleReport to the current web MyPage
            MyDocument.Draw(outputPdfPath)
        End Sub

        Public Shared Sub SetTemplate()
            ' Adds elements to the header template
            MyTemplate.Elements.Add(New Label(DateTime.Now.ToString("dd MMM yyyy, H:mm:ss EST"), 0, 0, 504, 12, Font.HelveticaBold, 12))
            MyTemplate.Elements.Add(New Label("Northwind Product List", 0, 0, 504, 12, Font.HelveticaBold, 12, TextAlign.Center))
            Dim MyPageNumLabel As PageNumberingLabel = New PageNumberingLabel("Page %%CP%% of %%TP%%", 0, 0, 504, 12, Font.HelveticaBold, 12, TextAlign.Right)
            MyTemplate.Elements.Add(MyPageNumLabel)
            MyTemplate.Elements.Add(New Label("Product", 2, 22, 236, 11, Font.TimesBold, 11))
            MyTemplate.Elements.Add(New Label("Qty Per Unit", 242, 22, 156, 11, Font.TimesBold, 11))
            MyTemplate.Elements.Add(New Label("Unit Price", 402, 22, 100, 11, Font.TimesBold, 11, TextAlign.Right))
            MyTemplate.Elements.Add(New Line(0, 36, 504, 36))

            ' Uncomment the line below to add a layout grid to each MyPage
            'MyTemplate.Elements.Add(New LayoutGrid())
        End Sub

        Public Shared Sub BuildDocument(ByVal MyDocument As Document, ByVal reader As XmlTextReader)
            ' Builds the PDF MyDocument with data from the XML Data
            AddNewPage(MyDocument)
            While reader.Read()
                If reader.NodeType = XmlNodeType.Element And reader.Name = "Products" Then
                    'Add current node to the MyDocument
                    AddRecord(MyDocument, reader)
                End If
            End While
        End Sub

        Public Shared Sub AddRecord(ByVal MyDocument As Document, ByVal reader As XmlTextReader)
            ' Adds a new MyPage to the MyDocument if needed
            If CurrentY > BodyBottom Then AddNewPage(MyDocument)

            ' Adds alternating background to MyDocument if needed
            If AlternateBG Then
                CurrentPage.Elements.Add(New Rectangle(0, CurrentY, 504, 18, Grayscale.Black, New WebColor("E0E0FF"), 0.0F))
            End If

            ' Adds Labels to the MyDocument with data from the current node
            CurrentPage.Elements.Add(New Label(reader.GetAttribute("ProductName"), 2, CurrentY + 2, 236, 11, Font.TimesRoman, 11))
            CurrentPage.Elements.Add(New Label(reader.GetAttribute("QuantityPerUnit"), 242, CurrentY + 2, 156, 11, Font.TimesRoman, 11))
            CurrentPage.Elements.Add(New Label(reader.GetAttribute("UnitPrice"), 402, CurrentY + 2, 100, 11, Font.TimesRoman, 11, TextAlign.Right))

            'Toggles alternating background
            AlternateBG = Not AlternateBG
            'Increments the current Y position on the MyPage
            CurrentY += 18
        End Sub

        Public Shared Sub AddNewPage(ByVal MyDocument As Document)
            'Adds a new MyPage to the MyDocument
            CurrentPage = New Page(MyPageDimensions)

            'Uncomment the line below to add a layout grid to the each MyPage
            'currentPage.Elements.Add(New LayoutGrid())

            CurrentY = BodyTop
            AlternateBG = False

            MyDocument.Pages.Add(CurrentPage)
        End Sub
    End Class
End Namespace

File: coresuite-dotnet-standard-cs/generator/SimpleXMLReport.cs
using ceTe.DynamicPDF;
using ceTe.DynamicPDF.PageElements;
using System;
using System.Xml;

namespace coresuite_dotnet_standard_cs.generator
{
    public class SimpleXMLReport
    {
        // Template for document elements
        public static Template template = new Template();

        // Page Dimensions of pages
        public static PageDimensions pageDimensions = new PageDimensions(PageSize.Letter, PageOrientation.Portrait, 54.0f);
        // Current page that elements are being added to
        public static Page currentPage = null;
        // Top Y coordinate for the body of the report
        public static float bodyTop = 38;
        // Bottom Y coordinate for the body of the report
        public static float bodyBottom = pageDimensions.Body.Bottom - pageDimensions.Body.Top;
        // Current Y coordinate where elements are being added
        public static float currentY = 0;

        // Used to control the alternating background
        public static bool alternateBG = false;

        public static void Run(string outputPdfPath)
        {
            // Create a document and set it's properties
            Document document = new Document
            {
                Creator = "SimpleXMLReport.aspx",
                Author = "ceTe Software",
                Title = "Simple XML Report"
            };
            XmlTextReader reader = new XmlTextReader(Util.GetResourcePath("Data/ProductList.xml"));

            // Adds elements to the header template
            SetTemplate();

            // Add the template to the document
            document.Template = template;

            // Builds the report
            BuildDocument(document, reader);

            // Outputs the SimpleReport to the current web page
            document.Draw(outputPdfPath);
        }

        public static void SetTemplate()
        {
            // Adds elements to the header template
            template.Elements.Add(new Label(DateTime.Now.ToString("dd MMM yyyy, H:mm:ss EST"), 0, 0, 504, 12, Font.HelveticaBold, 12));
            template.Elements.Add(new Label("Northwind Product List", 0, 0, 504, 12, Font.HelveticaBold, 12, TextAlign.Center));
            PageNumberingLabel pageNumLabel = new PageNumberingLabel("Page %%CP%% of %%TP%%", 0, 0, 504, 12, Font.HelveticaBold, 12, TextAlign.Right);
            template.Elements.Add(pageNumLabel);
            template.Elements.Add(new Label("Product", 2, 23, 236, 11, Font.TimesBold, 11));
            template.Elements.Add(new Label("Qty Per Unit", 242, 23, 156, 11, Font.TimesBold, 11));
            template.Elements.Add(new Label("Unit Price", 402, 23, 100, 11, Font.TimesBold, 11, TextAlign.Right));
            template.Elements.Add(new Line(0, 36, 504, 36));

            // Uncomment the line below to add a layout grid to each page
            //template.Elements.Add( new LayoutGrid() );
        }

        public static void BuildDocument(Document document, XmlTextReader reader)
        {
            // Builds the PDF document with data from the XML Data
            AddNewPage(document);
            while (reader.Read())
            {
                if (reader.NodeType == XmlNodeType.Element && reader.Name == "Products")
                {
                    //Add current node to the document
                    AddRecord(document, reader);
                }
            }
        }

        public static void AddRecord(Document document, XmlTextReader reader)
        {
            // Adds a new page to the document if needed
            if (currentY > bodyBottom) AddNewPage(document);

            // Adds alternating background to document if needed
            if (alternateBG)
            {
                currentPage.Elements.Add(new Rectangle(0, currentY, 504, 18, RgbColor.Black, new WebColor("E0E0FF"), 0.0F));
            }

            // Adds Labels to the document with data from the current node
            currentPage.Elements.Add(new Label(reader.GetAttribute("ProductName"), 2, currentY + 3, 236, 11, Font.TimesRoman, 11));
            currentPage.Elements.Add(new Label(reader.GetAttribute("QuantityPerUnit"), 242, currentY + 3, 156, 11, Font.TimesRoman, 11));
            currentPage.Elements.Add(new Label(reader.GetAttribute("UnitPrice"), 402, currentY + 3, 100, 11, Font.TimesRoman, 11, TextAlign.Right));

            // Toggles alternating background
            alternateBG = !alternateBG;

            // Increments the current Y position on the page
            currentY += 18;
        }

        public static void AddNewPage(Document document)
        {
            // Adds a new page to the document
            currentPage = new Page(pageDimensions);
            currentY = bodyTop;
            alternateBG = false;
            document.Pages.Add(currentPage);
        }
    }
}
